homeassistant/components/lamarzocco/lm_client.py

```diff
@@ -0,0 +1,147 @@
+"""La Marzocco Cloud API client."""
+from collections.abc import Callable
+import logging
+
+from lmcloud import LMCloud
+from lmcloud.exceptions import BluetoothConnectionFailed
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_MACHINE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LaMarzoccoClient(LMCloud):
+    """Keep data for La Marzocco entities."""
+
+    _bt_disconnected = False
+
+    def __init__(
+        self,
+        hass: HomeAssistant | None = None,
+        entry: ConfigEntry | None = None,
+        callback_websocket_notify: Callable[[], None] | None = None,
+    ) -> None:
+        """Initialise the LaMarzocco entity data."""
+        super().__init__(callback_websocket_notify=callback_websocket_notify)
+        self.entry = entry
+        self.hass = hass
+
+    async def connect(self) -> None:
+        """Connect to the machine."""
+        assert self.entry
```

Comment:
Assert should be avoided in production code as these can be disabled on runtime.
In which use case are both None?